$NetBSD$

Sprinkle DESTDIR dust.

Avoid installing backup~ files.

The standards.info provided by XEmacs conflicts with the one installed by
devel/autoconf, which is probably newer.

Prefix {c,e}tags to avoid conflicts.

Make TAR settable, to work around shortcomings in base tar(1)

--- Makefile.in.in.orig	2026-10-25 14:52:16.000000000 +0000
+++ Makefile.in.in
@@ -67,7 +67,7 @@ LC_ALL = C
 RM = rm -f
 MAKEPATH=./lib-src/make-path
 pwd = /bin/pwd
-TAR = tar
+TAR ?= tar
 
 ## ==================== Things `configure' Might Edit ====================
 
@@ -419,17 +419,17 @@ install-arch-dep: mkdir
 	   fi; \
 	   ${INSTALL_DATA} lib-src/config.values $(DESTDIR)${docdir}/config.values; \
 	   ${INSTALL_DATA} lib-src/DOC $(DESTDIR)${docdir}/DOC-`src/${PROGNAME} -sd`; \
-	   for subdir in `find ${archlibdir} -type d ! -name RCS ! -name SCCS ! -name CVS -print` ; \
+	   for subdir in `find $(DESTDIR)${archlibdir} -type d ! -name RCS ! -name SCCS ! -name CVS -print` ; \
 	     do (cd $${subdir} && $(RM) -r RCS CVS SCCS \#* *~) ; done ; \
 	else true; fi
 #if !defined (DUMP_IN_EXEC) && !defined(WIN32_NATIVE)
-	${INSTALL_DATA} src/${PROGNAME}.dmp ${archlibdir}/${PROGNAME}-`src/${PROGNAME} -sd`.dmp
+	${INSTALL_DATA} src/${PROGNAME}.dmp $(DESTDIR)${archlibdir}/${PROGNAME}-`src/${PROGNAME} -sd`.dmp
 #endif
 #ifdef WIN32_NATIVE
 	${INSTALL_PROGRAM} src/${PROGNAME} $(DESTDIR)${bindir}/${PROGNAME}
 	-chmod 0755 $(DESTDIR)${bindir}/${PROGNAME}
 #else
-	${INSTALL_PROGRAM} ${buildlispdir}/ellcc.elc $(DESDTIR)${bindir}/ellcc
+	${INSTALL_PROGRAM} ${buildlispdir}/ellcc.elc $(DESTDIR)${bindir}/ellcc
 	-chmod 0755 $(DESTDIR)${bindir}/ellcc
 # ifdef CYGWIN
 	${INSTALL_PROGRAM} src/${PROGNAME} $(DESTDIR)${bindir}/${PROGNAME}-${version}.exe
@@ -444,6 +444,8 @@ install-arch-dep: mkdir
 # endif /* CYGWIN */
 #endif /* WIN32_NATIVE */
 #ifdef HAVE_SHLIB
+	$(INSTALL_DATA) modules/auto-autoloads.el $(DESTDIR)$(moduledir)
+	$(INSTALL_DATA) modules/auto-autoloads.elc $(DESTDIR)$(moduledir)
 	$(INSTALL_DATA) modules/auto-autoloads.* $(DESTDIR)$(moduledir)
 #endif
 
@@ -479,15 +481,20 @@ install-arch-indep: mkdir info
 	    ${INSTALL_DATA} ${srcdir}/info/dir $(DESTDIR)${infodir}/dir ; \
 	  fi ; \
 	  for file in *.info* ; do \
+	    test $${file} = "standards.info" && continue ; \
 	    ${INSTALL_DATA} $${file} $(DESTDIR)${infodir}/$${file} ; \
 	    chmod 0644 $(DESTDIR)${infodir}/$${file}; \
 	  done ; \
 	fi
-	chmod 0755 ${lispdir}/ellcc.elc
+	chmod 0755 $(DESTDIR)${lispdir}/ellcc.elc
 	cd ${srcdir}/etc && \
 	  for page in xemacs etags ctags gnuserv gnuclient gnuattach gnudoit; do \
-	    ${INSTALL_DATA} ${srcdir}/etc/$${page}.1 $(DESTDIR)${mandir}/$${page}${manext} ; \
-	    chmod 0644 $(DESTDIR)${mandir}/$${page}${manext} ; \
+            if [ $${page} = etags ] || [ $${page} = ctags ] ; \
+            then \
+              ${BSD_INSTALL_MAN} ${srcdir}/etc/$${page}.1 $(DESTDIR)${mandir}/man1/xemacs-$${page}${manext} ; \
+            else \
+              ${BSD_INSTALL_MAN} ${srcdir}/etc/$${page}.1 $(DESTDIR)${mandir}/man1/$${page}${manext} ; \
+            fi \
 	  done
 	@echo "If you would like to save approximately 4M of disk space, do"
 	@echo "make gzip-el"
@@ -506,7 +513,7 @@ gzip-el:
 mkdir: FRC.mkdir
 	for dir in \
 	    ${COPYDESTS} ${docdir} ${infodir} ${archlibdir} \
-	    ${mandir} ${bindir} ${datadir} ${libdir} \
+	    ${mandir}/man1 ${bindir} ${datadir} ${libdir} \
 #ifdef HAVE_SHLIB
 	    ${moduledir} ${sitemoduledir} \
 #endif
